app/src/ui/branches/branch-list-item.tsx

```diff
@@ -18,10 +20,51 @@ interface IBranchListItemProps {
 
   /** The characters in the branch name to highlight */
   readonly matches: IMatches
+
+  /** Specifies whether the branch is local */
+  readonly isLocal: boolean
+
+  readonly onRenameBranch?: (branchName: string) => void
+
+  readonly onDeleteBranch?: (branchName: string) => void
 }
 
 /** The branch component. */
 export class BranchListItem extends React.Component<IBranchListItemProps, {}> {
+  private onContextMenu = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.preventDefault()
+
+    /*
+      There are multiple instances in the application where a branch list item
+      is rendered. We only want to be able to rename or delete them on the
+      branch dropdown menu. Thus, other places simply will not provide these
+      methods, such as the merge and rebase logic.
+    */
+    const { onRenameBranch, onDeleteBranch, name, isLocal } = this.props
+    if (onRenameBranch === undefined && onDeleteBranch === undefined) {
+      return
+    }
+
+    const items: Array<IMenuItem> = []
+
+    if (onRenameBranch !== undefined) {
+      items.push({
+        label: 'Rename...',
```

Comment:
Nit: prefer the use of `…` over `...`?
```suggestion
        label: 'Rename…',
```